### PR TITLE
Add darken/lighten variations to theme colors

### LINKED ISSUE #1765: [Enhancement] `darken-x` and `lighten-x` for theme colors
The classes `darken-x` and `lighten-x` for [these colors](https://vuetifyjs.com/style/colors) are very helpful. I wish they also worked with `primary`, `secondary`, and such theme colors. Same with the text versions.

Since these are supposedly the two most important colors in the palette, it'd be great if we have finer control over their shades. Currently, it requires a lot of custom classes to achieve these, at least for people like me who aren't very comfortable with editing the default stylus variables/values.

### PR DESCRIPTION
**Pull request by KaelWD**
- Adds `darken-x` and `lighten-x` classes for theme colors
 - New global config: `$vuetify.options`
 - `$vuetify.options.themeVariations` can be set to an array to limit which colors get variations. The default is `['primary', 'secondary', 'accent']`
 - `$vuetify.options.minifyTheme` can be set to a function that will minify the theme stylesheet. This is only useful for SSR when you want to reduce your page size as much as possible
 - `$vuetify.options.themeCache` takes an object that will cache the generated stylesheet (again, SSR)

Closes #1765

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

File: src/util/colorUtils.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  let rgb
+
+  if (typeof color === 'number') {
+    rgb = color
+  } else if (typeof color === 'string') {
+    // TODO: more string formats
+    const c = color.substring(1)
+    rgb = parseInt(c, 16)
+  } else {
+    throw new TypeError(`Colors can only be numbers or strings, recieved ${color.constructor.name} instead`)
```
Comment (johnleider) [inline src/util/colorUtils.js]: refactor this to use the functionality @jacekkarczmarczyk just [created](https://github.com/vuetifyjs/vuetify/pull/2920)

File: src/util/colorUtils.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  }
+
+  if (rgb < 0) {
+    throw new RangeError(`Colors cannot be negative: '${color}'`)
```
Comment (johnleider) [inline src/util/colorUtils.js]: refactor this to use the functionality @jacekkarczmarczyk just [created](https://github.com/vuetifyjs/vuetify/pull/2920)

File: src/util/colorUtils.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  } else if (rgb > 0xffffff) {
+    throw new RangeError(`'${color}' is not a valid rgb color`)
```
Comment (johnleider) [inline src/util/colorUtils.js]: refactor this to use the functionality @jacekkarczmarczyk just [created](https://github.com/vuetifyjs/vuetify/pull/2920)

File: src/util/colorUtils.js (hunk 1 above)
Comment (KaelWD) [reply src/util/colorUtils.js]: This one should still be an error, otherwise there'll be a more cryptic on later on.

**Pull request merged**